doc/Requirements Specification/Parts/Non-Functional Requirements.lyx

```diff
@@ -430,5 +430,274 @@ gls{SoMoX}
 .
 \end_layout
 
+\begin_layout Subsection
+GUI model
+\end_layout
+
+\begin_layout Standard
+The user has multiple options to launch the analysis:
+\end_layout
+
+\begin_layout Enumerate
+To analyse the whole project, there is an entry 
+\begin_inset Quotes eld
+\end_inset
+
+Analyse with Beagle
+\begin_inset Quotes erd
+\end_inset
+
+ in the context menu of the 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+lstinline{.repository}
+\end_layout
+
+\end_inset
+
+ or 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+lstinline{.repository_diagram}
+\end_layout
+
+\end_inset
+
+ file in eclipse.
+\end_layout
+
+\begin_layout Enumerate
+To analyse a single component, there is an entry 
+\begin_inset Quotes eld
+\end_inset
+
+Analyse with Beagle
+\begin_inset Quotes erd
+\end_inset
+
+ in the context menu of each 
+\begin_inset ERT
+status open
+
+\begin_layout Plain Layout
+
+
+\backslash
+gls{component}
+\end_layout
+
+\end_inset
+
+ in the repository diagram, where an analysis with Beagle is possible.
+ Otherwise the context menu entry will be shown greyed and a description
+ why the analysis is not possible appears, when the users tries to start
+ the analysis.
```

Comment:
These constraints also apply if launching Beagle for a whole project or a single internal action.